discretisedfield/tests/test_mesh.py

```diff
@@ -1688,3 +1688,183 @@ def test_sel_subregions():
 
     # cell boundary and cell midpoint return the same selection
     assert mesh.sel(x0=6) == mesh.sel(x0=7)
+
+
+def test_fftn_mesh():
+
+    # 3D mesh with even number of cells
+    p1 = (-10, -10, -5)
+    p2 = (10, 10, 5)
+    cell = (1, 1, 1)
+    mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+
+    mesh_fft = mesh.fftn()
+
+    assert mesh_fft.region.ndim == 3
+    assert mesh_fft.region.dims == ("k_x", "k_y", "k_z")
+    assert mesh_fft.region.units == ("(m)^-1", "(m)^-1", "(m)^-1")
+    assert np.array_equal(mesh_fft.n, (20, 20, 10))
+    assert np.allclose(mesh_fft.cell, (0.05, 0.05, 0.1))
+    assert np.allclose(mesh_fft.region.pmin, (-0.525, -0.525, -0.55))
+    assert np.allclose(mesh_fft.region.pmax, (0.475, 0.475, 0.45))
+
+    mesh_fft = mesh.fftn(rfft=True)
+
+    assert mesh_fft.region.ndim == 3
+    assert mesh_fft.region.dims == ("k_x", "k_y", "k_z")
+    assert mesh_fft.region.units == ("(m)^-1", "(m)^-1", "(m)^-1")
+    assert np.array_equal(mesh_fft.n, (20, 20, 6))
+    assert np.allclose(mesh_fft.cell, (0.05, 0.05, 0.1))
+    assert np.allclose(mesh_fft.region.pmin, (-0.525, -0.525, -0.05))
+    assert np.allclose(mesh_fft.region.pmax, (0.475, 0.475, 0.55))
+
+    mesh_fft = mesh.fftn().ifftn()
+    assert mesh_fft.region.ndim == 3
+    assert np.array_equal(mesh_fft.n, (20, 20, 10))
+    assert np.allclose(mesh_fft.cell, (1, 1, 1))
+    assert mesh_fft.region.units == ("m", "m", "m")
+    assert mesh_fft.region.dims == ("x", "y", "z")
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
+
+    mesh_fft = mesh.fftn(rfft=True).ifftn(rfft=True)
+    assert mesh_fft.region.ndim == 3
+    assert np.array_equal(mesh_fft.n, (20, 20, 10))
+    assert np.allclose(mesh_fft.cell, (1, 1, 1))
+    assert mesh_fft.region.units == ("m", "m", "m")
+    assert mesh_fft.region.dims == ("x", "y", "z")
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
+
+    mesh_fft = mesh.fftn(rfft=True).ifftn(rfft=True, shape=mesh.n)
+    assert mesh_fft.region.ndim == 3
+    assert np.array_equal(mesh_fft.n, (20, 20, 10))
+    assert np.allclose(mesh_fft.cell, (1, 1, 1))
+    assert mesh_fft.region.units == ("m", "m", "m")
+    assert mesh_fft.region.dims == ("x", "y", "z")
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
```

Comment:
Apart from the checks for `units` and `dims` you could use `mesh.allclose(mesh_fft)`.